api/middlewares/auth.js

```diff
@@ -31,22 +31,37 @@ exports.checkCredentials = async (req, res, next) => {
   });
 };
 
-exports.fetchVoter = async (req, res, next) => {
-  const voterID = req.query.voterID;
-
-  Voters.findOne({ voterID: voterID }).exec(async (error, voterData) => {
+exports.verifyVoter = async (req, res, next) => {
+  let query;
+  if (req.query.voterID) {
+    query = {
+      voterID: req.query.voterID,
+    };
+  } else {
+    query = {
+      phone: req.body.phone,
+    };
+  }
+  console.log(query);
+  Voters.findOne(query).exec(async (error, voterData) => {
     if (error) {
       // some error occured
       return res.status(400).json({ error });
     }
     if (voterData) {
-      // VoterID is correct checking for password
-      req.phone = voterData.phone;
-      req.district = voterData.pinCode;
-      req.hasRegistered = voterData.hasRegistered;
```

Comment:
similar doubt

Reply:
They are there but I have moved them in an if check. PTAL